SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialAuthorityTest/SpatialAuthorityTest.cpp

```diff
@@ -223,59 +287,103 @@ void ASpatialAuthorityTest::PrepareTest()
 			FinishStep();
 		});
 
+		AddStepFromDefinition(NonReplicatedVerifyAuthorityStepDefinition, FWorkerDefinition::Server(1));
+
+		AddStepFromDefinition(NonReplicatedVerifyOthersStepDefinition, FWorkerDefinition::AllWorkers);
+
+		AddStepFromDefinition(NonReplicatedDestroyStepDefinition, FWorkerDefinition::Server(1));
+	}
+
+	// Replicated Dynamic Actor Spawned On Border from Server 2. Server 2 should keep Authority.
+	{
+		AddStep(TEXT("Replicated Dynamic Actor Spawned On Border From Server 2 - Spawn"), FWorkerDefinition::Server(2), nullptr, [this]() {
+			ASpatialAuthorityTestReplicatedActor* Actor =
+				GetWorld()->SpawnActor<ASpatialAuthorityTestReplicatedActor>(BorderPosition, FRotator::ZeroRotator);
+			CrossServerSetDynamicReplicatedActor(Actor);
+			FinishStep();
+		});
+
 		AddStep(
-			TEXT("Non-replicated Dynamic Actor - Verify Authority on Server 1"), FWorkerDefinition::Server(1), nullptr, nullptr,
+			TEXT("Replicated Dynamic Actor Spawned On Border From Server 2 - Verify Server 2 Has Authority"), FWorkerDefinition::AllWorkers,
+			nullptr, nullptr,
 			[this](float DeltaTime) {
-				// Not replicated so OnAuthorityGained() is not called.
-				if (VerifyTestActor(DynamicNonReplicatedActor, 1, 1, 0, 0))
+				Timer -= DeltaTime;
+				if (Timer <= 0)
 				{
-					FinishStep();
+					const FWorkerDefinition& LocalWorkerDefinition = GetLocalFlowController()->WorkerDefinition;
+					if (LocalWorkerDefinition.Type == ESpatialFunctionalTestWorkerType::Server && LocalWorkerDefinition.Id == 2)
+					{
+						if (VerifyTestActor(DynamicReplicatedActor, 2, 2, 1, 0))
+						{
+							FinishStep();
+						}
+					}
+					else
+					{
+						if (VerifyTestActor(DynamicReplicatedActor, 0, 0, 0, 0))
+						{
+							FinishStep();
+						}
+					}
 				}
 			},
 			5.0f);
 
-		AddStep(TEXT("Non-replicated Dynamic Actor - Verify Dynamic Actor doesn't exist on others"), FWorkerDefinition::AllWorkers, nullptr,
-				nullptr, [this](float DeltaTime) {
+		AddStepFromDefinition(ReplicatedDestroyStepDefinition, FWorkerDefinition::Server(2));
+	}
+
+	// Replicated Dynamic Actor Spawned On Border from Server 1. Server 1 should keep Authority.
+	// servers
+	{
+		AddStep(TEXT("Replicated Dynamic Actor Spawned On Border From Server 1 - Spawn"), FWorkerDefinition::Server(1), nullptr, [this]() {
+			ASpatialAuthorityTestReplicatedActor* Actor =
+				GetWorld()->SpawnActor<ASpatialAuthorityTestReplicatedActor>(BorderPosition, FRotator::ZeroRotator);
```

Comment:
Oh - so it seems BorderPosition is in server 1's auth area?

Reply:
I changed it to a 2x2 load balancing strategy and all workers pass through the border position of (0,0,0), as shown below:
![image](https://user-images.githubusercontent.com/64085832/102065388-2a453480-3e01-11eb-8e40-cc0a63bcf5cc.png)

Reply:
Sure - but the test case here and the one above both spawn the actor at the same position - BorderPosition .
Shouldn't the load balancer resolve that to the same server in both instances (either 1 or 2).
It looks like in the first test, it stays on server 2, and the second, it stays on server 1.

Reply:
Investigated this further and the issue is with the test: initially, the actor will be on the server that spawned it which is what is checked so far but an additional check is also needed to show that in all (0,0,0) cases the actor will migrate to worker 4.